### PR TITLE
Fix 17480 - Update file contents of the release archives & link to them on downloads.html

### PR DESCRIPTION
**Pull request by wilzbach**
- The issue is about the fact that we never mention that DUB is part of the release archives
- I tried to update the file contents archive and link to them (img + text)
- See below for the actual file listings from the archives

CC @MartinNowak @CyberShadow

File archives
----------------

- 2.074.1
- for better readability I removed the hits from other directories (the default configs of every OS are part of nearly all archives)

```
> als dmd.2.074.1.windows.7z | grep bin/
2017-05-31 17:29:06 ....A         1011               dmd2/windows/bin/README.TXT
2017-05-31 17:41:22 ....A      3772330               dmd2/windows/bin/d.chm
2017-05-31 17:28:32 ....A         4630               dmd2/windows/bin/sc.ini
2017-05-31 17:29:06 ....A       215274               dmd2/windows/bin/windbg.hlp
2017-05-31 17:41:50 ....A       540188      4129070  dmd2/windows/bin/ddemangle.exe
2017-05-31 17:29:08 ....A        85776               dmd2/windows/bin/dm.dll
2017-05-31 17:42:18 ....A      2300956               dmd2/windows/bin/dman.exe
2017-05-31 17:37:44 ....A      3165724               dmd2/windows/bin/dmd.exe
2017-05-31 17:43:06 ....A      8267908               dmd2/windows/bin/dub.exe
2017-05-31 17:42:06 ....A       891420               dmd2/windows/bin/dustmite.exe
2017-05-31 17:29:06 ....A       150288               dmd2/windows/bin/eecxxx86.dll
2017-05-31 17:29:08 ....A        80656               dmd2/windows/bin/emx86.dll
2017-05-31 17:29:08 ....A        65564               dmd2/windows/bin/lib.exe
2017-05-31 17:29:06 ....A       451072               dmd2/windows/bin/libcurl.dll
2017-05-31 17:29:08 ....A       223260               dmd2/windows/bin/link.exe
2017-05-31 17:29:08 ....A        49692               dmd2/windows/bin/make.exe
2017-05-31 17:29:06 ....A       275216               dmd2/windows/bin/mspdb41.dll
2017-05-31 17:41:36 ....A       796700               dmd2/windows/bin/rdmd.exe
2017-05-31 17:29:06 ....A        32768               dmd2/windows/bin/replace.exe
2017-05-31 17:29:08 ....A        57616               dmd2/windows/bin/shcv.dll
2017-05-31 17:29:08 ....A        59932               dmd2/windows/bin/shell.exe
2017-05-31 17:29:08 ....A         8976               dmd2/windows/bin/tlloc.dll
2017-05-31 17:29:06 ....A       559888               dmd2/windows/bin/windbg.exe
```

```
> als dmd.2.074.1.osx.tar.xz | grep bin/
drwxrwxr-x builder/builder    0 2017-06-01 03:50 dmd2/osx/bin/
-rwxr-xr-x builder/builder 6107444 2017-05-31 18:11 dmd2/osx/bin/dmd
-rwxr-xr-x builder/builder   75888 2014-10-11 18:08 dmd2/osx/bin/dumpobj
-rwxr-xr-x builder/builder 5819112 2017-05-31 18:14 dmd2/osx/bin/dustmite
-rwxr-xr-x builder/builder 3816016 2017-05-31 18:13 dmd2/osx/bin/rdmd
-rwxr-xr-x builder/builder 2832204 2017-05-31 18:14 dmd2/osx/bin/ddemangle
-rw-rw-r-- builder/builder      72 2017-05-31 16:25 dmd2/osx/bin/dmdx.conf
-rwxr-xr-x builder/builder 13122584 2017-05-31 18:15 dmd2/osx/bin/dub
-rwxr-xr-x builder/builder   133412 2014-10-11 18:08 dmd2/osx/bin/obj2asm
-rwxr-xr-x builder/builder    42924 2014-10-11 18:08 dmd2/osx/bin/shell
-rw-rw-r-- builder/builder       96 2017-05-31 16:25 dmd2/osx/bin/dmd.conf
-rwxr-xr-x builder/builder  5090432 2017-05-31 18:14 dmd2/osx/bin/dman
```

```
> als dmd.2.074.1.freebsd-64.tar.xz | grep 'bin.*/'
drwxrwxr-x builder/builder        0 2017-06-01 03:56 dmd2/freebsd/bin64/
-rwxr-xr-x builder/builder  6228318 2017-06-01 01:41 dmd2/freebsd/bin64/dmd
-rwxr-xr-x builder/builder  5309115 2017-06-01 01:43 dmd2/freebsd/bin64/dustmite
-rwxr-xr-x builder/builder  3691552 2017-06-01 01:43 dmd2/freebsd/bin64/rdmd
-rwxr-xr-x builder/builder  2765952 2017-06-01 01:43 dmd2/freebsd/bin64/ddemangle
-rwxr-xr-x builder/builder 12334055 2017-06-01 01:44 dmd2/freebsd/bin64/dub
-rw-rw-r-- builder/builder      235 2017-05-31 23:25 dmd2/freebsd/bin64/dmd.conf
-rwxr-xr-x builder/builder  5329518 2017-06-01 01:43 dmd2/freebsd/bin64/dman
drwxrwxr-x builder/builder        0 2017-06-01 03:56 dmd2/freebsd/bin32/
-rwxr-xr-x builder/builder    54204 2014-10-14 23:15 dmd2/freebsd/bin32/dumpobj
-rwxr-xr-x builder/builder    92268 2014-10-14 23:15 dmd2/freebsd/bin32/obj2asm
-rwxr-xr-x builder/builder    37608 2014-10-14 23:15 dmd2/freebsd/bin32/shell
-rw-rw-r-- builder/builder      235 2017-05-31 23:25 dmd2/freebsd/bin32/dmd.conf
```

```
>  als dmd.2.074.1.linux.tar.xz | grep 'bin.*/'
drwxrwxr-x builder/builder        0 2017-06-01 03:44 dmd2/linux/bin64/
-rwxr-xr-x builder/builder  6350741 2017-05-31 23:46 dmd2/linux/bin64/dmd
-rwxr-xr-x builder/builder    67984 2014-10-14 23:23 dmd2/linux/bin64/dumpobj
-rwxr-xr-x builder/builder  5341595 2017-05-31 23:48 dmd2/linux/bin64/dustmite
-rwxr-xr-x builder/builder  3732332 2017-05-31 23:48 dmd2/linux/bin64/rdmd
-rwxr-xr-x builder/builder  2795460 2017-05-31 23:48 dmd2/linux/bin64/ddemangle
-rw-rw-r-- builder/builder      417 2017-05-31 23:25 dmd2/linux/bin64/README.TXT
-rwxr-xr-x builder/builder 12517651 2017-05-31 23:49 dmd2/linux/bin64/dub
-rwxr-xr-x builder/builder   123992 2014-10-14 23:23 dmd2/linux/bin64/obj2asm
-rw-rw-r-- builder/builder      241 2017-05-31 23:25 dmd2/linux/bin64/dmd.conf
-rwxr-xr-x builder/builder  5365831 2017-05-31 23:48 dmd2/linux/bin64/dman
drwxrwxr-x builder/builder        0 2017-06-01 03:44 dmd2/linux/bin32/
-rwxr-xr-x builder/builder  5424997 2017-05-31 23:33 dmd2/linux/bin32/dmd
-rwxr-xr-x builder/builder    63040 2014-10-14 23:23 dmd2/linux/bin32/dumpobj
-rwxr-xr-x builder/builder  4869208 2017-05-31 23:39 dmd2/linux/bin32/dustmite
-rwxr-xr-x builder/builder  3298357 2017-05-31 23:38 dmd2/linux/bin32/rdmd
-rwxr-xr-x builder/builder  2488103 2017-05-31 23:39 dmd2/linux/bin32/ddemangle
-rw-rw-r-- builder/builder      417 2017-05-31 23:25 dmd2/linux/bin32/README.TXT
-rwxr-xr-x builder/builder 11276690 2017-05-31 23:39 dmd2/linux/bin32/dub
-rwxr-xr-x builder/builder   114544 2014-10-14 23:23 dmd2/linux/bin32/obj2asm
-rw-rw-r-- builder/builder      235 2017-05-31 23:25 dmd2/linux/bin32/dmd.conf
-rwxr-xr-x builder/builder  4406870 2017-05-31 23:39 dmd2/linux/bin32/dman
```

### CODE REVIEW

File: dcompiler.dd [hunk 1]
```diff
@@ -71,91 +71,192 @@ $(H2 $(LNAME2 files, Files))
...
+    $(DD D manual lookup tool)
+    )
+
     $(DT $(D $(DMDDIR)/linux/bin/dmd)
     $(DD D compiler executable)
     )
 
+    $(DT $(D $(DMDDIR)/linux/bin/$(DMD_CONF))
+    $(DD Global compiler settings (copy to $(D /etc/dmd.conf)))
+    )
```
Comment (wilzbach) [inline dcompiler.dd]: As everything else is alphabetically sorted, I tried to stay consistent.

File: dcompiler.dd [hunk 2]
```diff
@@ -71,91 +71,192 @@ $(H2 $(LNAME2 files, Files))
...
-    $(DT $(D $(DMDDIR)/osx/bin/$(SHELL))
-    $(DD Simple command line shell)
+    $(DT $(D $(DMDDIR)/osx/bin/$(RDMD))
+    $(DD D build tool for script-like D code execution)
     )
 
-    $(DT $(D $(DMDDIR)/osx/bin/$(DMD_CONF))
-    $(DD Global compiler settings (copy to $(D /etc/dmd.conf)))
+    $(DT $(D $(DMDDIR)/osx/bin/$(SHELL))
+    $(DD Simple command line shell)
```
Comment (wilzbach) [inline dcompiler.dd]: I wonder whether anyone actually uses this

File: doc.ddoc [hunk 3]
```diff
@@ -14,9 +14,6 @@ LAYOUT=$3
 _=
 
 SFINAE=$(GLOSSARY2 sfinae, $(ACRONYM SFINAE, Substitution Failure Is Not An Error))
-OBJ2ASM=$(HTTP digitalmars.com/ctg/obj2asm.html, obj2asm)
-DUMPOBJ=$(HTTP digitalmars.com/ctg/dumpobj.html, dumpobj)
-SHELL=$(HTTP digitalmars.com/ctg/shell.html, shell)
```
Comment (wilzbach) [inline doc.ddoc]: These global variables were only used in `dcompiler.dd`

File: dcompiler.dd [hunk 4]
```diff
@@ -71,91 +71,192 @@ $(H2 $(LNAME2 files, Files))
...
+    $(DD OPTLINK)
+    )
+
+    $(DT $(D $(DMDDIR)\windows\bin\$(LINK2 http://www.digitalmars.com/ctg/make.html, make.exe))
+    $(DD Digitalmars Make)
+    )
+
+    $(DT $(D $(DMDDIR)\windows\bin\$(OBJ2ASM))
+    $(DD Elf file disassembler)
+    )
```
Comment (CyberShadow) [inline dcompiler.dd]: Not on Windows

File: dcompiler.dd [hunk 5]
```diff
@@ -71,91 +71,192 @@ $(H2 $(LNAME2 files, Files))
...
+
+    $(DT $(D $(DMDDIR)\windows\bin\replace.exe)
+    $(DD Find/replace text in files)
+    )
```
Comment (CyberShadow) [inline dcompiler.dd]: Ditto

**Comment by CyberShadow** [comment]
Not sure about making the platform name a link to the documentation. I'd think that being on the download page, clicking it would take me to all Windows downloads.

**Comment by wilzbach** [comment]
> Not sure about making the platform name a link to the documentation. I'd think that being on the download page, clicking it would take me to all Windows downloads.

Okay - I cherry-picked this commit into [another PR](https://github.com/dlang/dlang.org/pull/1697)

**Comment by CyberShadow** [comment]
http://dtest.dlang.io/diff/website-5bebaa7421e3cdb543c3d1f90c6856902a3fc561-267e02604b2ebb60d117e4f6c6dc779a/website-d659f995136d8cb6982bdf144a5bf69bca80b04f-b7c400e1bf1ccc549a8a1541e257bf5b/web/dmd-freebsd.html

```diff
+    <dt><span class="d_inlinecode donthyphenate notranslate">dmd2/freebsd/bin/<a href="rdmd.html">rdmd<!--UNDEFINED MACRO: "BIN_EXT"--></a></span>
+    <dd>D build tool for script-like D code execution</dd>
```

`UNDEFINED MACRO: "BIN_EXT"` doesn't look right.

**Comment by CyberShadow** [comment]
Also would be nice to catch those in CI.

**Comment by wilzbach** [comment]
> Also would be nice to catch those in CI.

Yeah we really should do this.:

```
> grep -r "UNDEFINED MACRO" web
web/dmd-linux.html:    <dt><span class="d_inlinecode donthyphenate notranslate">/linux/bin/<a href="https://github.com/dlang/dub">dub<!--UNDEFINED MACRO: "BINEXT"--></a></span>
web/dmd-linux.html:    <dt><span class="d_inlinecode donthyphenate notranslate">/linux/bin/<a href="http://digitalmars.com/ctg/dumpobj.html">dumpobj<!--UNDEFINED MACRO: "BINEXT"--></a></span>
web/dmd-linux.html:    <dt><span class="d_inlinecode donthyphenate notranslate">/linux/bin/<a href="https://github.com/CyberShadow/DustMite">dustmite<!--UNDEFINED MACRO: "BINEXT"--></a></span>
web/dmd-linux.html:    <dt><span class="d_inlinecode donthyphenate notranslate">/linux/bin/<a href="http://digitalmars.com/ctg/obj2asm.html">obj2asm<!--UNDEFINED MACRO: "BINEXT"--></a></span>
web/dmd-linux.html:    <dt><span class="d_inlinecode donthyphenate notranslate">/linux/bin/<a href="rdmd.html">rdmd<!--UNDEFINED MACRO: "BIN_EXT"--></a></span>
web/phobos-prerelease/std_algorithm.html:interfaces.  See also <!--UNDEFINED MACRO: "MREF_ALTTEF"--> and
web/phobos-prerelease/std_math.html: <!--UNDEFINED MACRO: "TABLE_DOMRG"-->
web/phobos-prerelease/std_math.html: <!--UNDEFINED MACRO: "TABLE_DOMRG"-->
web/phobos-prerelease/std_json.html:Set the <!--UNDEFINED MACRO: "specialFloatLiterals"--> flag is set in <span class="d_inlinecode donthyphenate notranslate"><code class="ddoc_param">options</code></span> to encode NaN/Infinity as strings.</div>
web/phobos-prerelease/std_datetime_date.html:       value of <span class="d_inlinecode donthyphenate notranslate">T</span> as <!--UNDEFINED MACRO: "Unqual"-->.
web/phobos-prerelease/std_datetime_date.html:       value of <span class="d_inlinecode donthyphenate notranslate">T</span> as <!--UNDEFINED MACRO: "Unqual"-->.
web/phobos-prerelease/std_random.html: <table class="book"><caption><!--UNDEFINED MACRO: "TEXTWITHCOMMAS"--></caption>  <tr><th scope="col">bits</th> <th scope="col">period</th></tr>
web/phobos-prerelease/std_socket.html:<dd><div class="summary">Wait for a socket to change status. A wait <code class="ddoc_param">timeout</code> of <!--UNDEFINED MACRO: "Duration"--> or
web/phobos-prerelease/object.html: takes a <!--UNDEFINED MACRO: "PARAM"--> delegate.</div>
web/phobos-prerelease/object.html: <!--UNDEFINED MACRO: "PARAM"--> delegate to avoid GC allocations, which cannot be
web/phobos-prerelease/std_experimental_allocator_building_blocks_free_list.html:<div class="description"><span class="d_inlinecode donthyphenate notranslate">expand</span> is defined to forward to <!--UNDEFINED MACRO: "ParentAllocator"--> (it must be also
web/dmd-freebsd.html:    <dt><span class="d_inlinecode donthyphenate notranslate">dmd2/freebsd/bin/<a href="rdmd.html">rdmd<!--UNDEFINED MACRO: "BIN_EXT"--></a></span>
web/faq.html:	<!--UNDEFINED MACRO: "Currently"-->
web/dmd-osx.html:    <dt><span class="d_inlinecode donthyphenate notranslate">dmd2/osx/bin/<a href="rdmd.html">rdmd<!--UNDEFINED MACRO: "BIN_EXT"--></a></span>
web/dmd-windows.html:    <dt><span class="d_inlinecode donthyphenate notranslate">dmd2\windows\bin\<a href="rdmd.html">rdmd<!--UNDEFINED MACRO: "BIN_EXT"--></a></span>
web/forum-template.html:<div class="smallprint" id="copyright"><!--UNDEFINED MACRO: "COPYRIGHT"--></div>
```

**Comment by wilzbach** [comment]
> Yeah we really should do this.:

After these PRs, we could enable a simple `grep` check :)

- [ ] https://github.com/dlang/phobos/issues/5486
- [ ] https://github.com/dlang/druntime/issues/1840
- [ ] https://github.com/dlang/dlang.org/pull/1707

**Comment by CyberShadow** [comment]
Awesome

**Comment by CyberShadow** [comment]
Noticed another regression in the diff:

```diff
-    <span class="d_inlinecode donthyphenate notranslate">c:\dmd2\bin\sc.ini</span>, then <b>%@P%</b> will be replaced with
-    <span class="d_inlinecode donthyphenate notranslate">c:\dmd2\bin</span>, and the above <span class="d_inlinecode donthyphenate notranslate">sc.ini</span> will be
+    <span class="d_inlinecode donthyphenate notranslate">c:dmd2\bin\sc.ini</span>, then <b>%@P%</b> will be replaced with
+    <span class="d_inlinecode donthyphenate notranslate">c:dmd2\bin</span>, and the above <span class="d_inlinecode donthyphenate notranslate">sc.ini</span> will be
```

Backslashes got eaten.

File: dcompiler.dd (hunk 4 above)
Comment (CyberShadow) [inline dcompiler.dd]: Even though I now see this + is just a reordering, this is still wrong, isn't it? There is no `obj2asm.exe` in the Windows distribution, and seeing as how it's described as an "Elf file disassembler", it wouldn't make sense to include it. (Also, it's ELF, not Elf.)

**Comment by wilzbach** [comment]
> After these PRs, we could enable a simple grep check :)

-> https://github.com/dlang/dlang.org/pull/1713

**Comment by CyberShadow** [comment]
There are still missing backslashes in the diff:

```diff

@@ -600,7 +637,7 @@ C:\&gt;unzip dmc.zip
     way to set <b>LIB</b> would be:
     </p>
 
-<pre class="console notranslate">set LIB=\dmd2\lib;\dm\lib
+<pre class="console notranslate">set LIB=dmd2\lib;\dm\lib
 </pre>
 
     <p>If the <b>dmd</b> command is used to both compile and link
@@ -634,7 +671,7 @@ C:\&gt;unzip dmc.zip
     <dd>The linker uses <b>LIB</b> to search for library files. For D, it will
     normally be set to:</dd>
 
-<pre class="console notranslate">set LIB=\dmd2\lib;\dm\lib
+<pre class="console notranslate">set LIB=dmd2\lib;\dm\lib
 </pre>
 
     <dt><b>LINKCMD</b></dt>
@@ -711,16 +748,16 @@ DDOCFILE=mysettings.ddoc
 
     <p>The <b>%@P%</b> is replaced with the path to <span class="d_inlinecode donthyphenate notranslate">sc.ini</span>.
     Thus, if the fully qualified file name <span class="d_inlinecode donthyphenate notranslate">sc.ini</span> is
-    <span class="d_inlinecode donthyphenate notranslate">c:\dmd2\bin\sc.ini</span>, then <b>%@P%</b> will be replaced with
-    <span class="d_inlinecode donthyphenate notranslate">c:\dmd2\bin</span>, and the above <span class="d_inlinecode donthyphenate notranslate">sc.ini</span> will be
+    <span class="d_inlinecode donthyphenate notranslate">c:dmd2\bin\sc.ini</span>, then <b>%@P%</b> will be replaced with
+    <span class="d_inlinecode donthyphenate notranslate">c:dmd2\bin</span>, and the above <span class="d_inlinecode donthyphenate notranslate">sc.ini</span> will be
     interpreted as:
     </p>
 
 <pre class="scini notranslate">[Environment]
-LIB="c:\dmd2\bin\..\lib";\dm\lib
+LIB="c:dmd2\bin\..\lib";\dm\lib
 
-DFLAGS="-Ic:\dmd2\bin\..\src\phobos" "-Ic:\dmd2\bin\..\src\druntime\import"
-LINKCMD="c:\dmd2\bin\..\..\dm\bin"
+DFLAGS="-Ic:dmd2\bin\..\src\phobos" "-Ic:dmd2\bin\..\src\druntime\import"
+LINKCMD="c:dmd2\bin\..\..\dm\bin"
 DDOCFILE=mysettings.ddoc
 </pre>
 ```

File: dcompiler.dd [hunk 6]
```diff
@@ -71,91 +71,184 @@ $(H2 $(LNAME2 files, Files))
...
+    $(DD Global compiler settings)
+    )
+
     $(DT $(D $(DMDDIR)\windows\bin\$(LINK2 http://www.digitalmars.com/ctg/shell.html, shell.exe))
     $(DD Simple command line shell)
     )
 
-    $(DT $(D $(DMDDIR)\windows\bin\sc.ini)
-    $(DD Global compiler settings)
+    $(DT $(D $(DMDDIR)\windows\bin\$(LINK2 http://www.digitalmars.com/d/2.0/windbg.html, windb.exe))
```
Comment (CyberShadow) [inline dcompiler.dd]: windbg, not windb

**Comment by wilzbach** [comment]
> Hopefully last one :)

Finally managed to catch up with the queue / this PR - fixed (and squashed due to the long period).

**Comment by CyberShadow** [comment]
Thanks!

**Pull request merged**